rama-http-backend/src/server/layer/upgrade/service.rs

```diff
@@ -120,9 +119,7 @@ where
                         async move {
                             match rama_http::io::upgrade::handle_upgrade(&req).await {
                                 Ok(mut upgraded) => {
-                                    upgraded
-                                        .extensions_mut()
-                                        .set_parent_extensions(Arc::new(req.extensions().clone()));
+                                    upgraded.extensions_mut().extend(req.extensions().clone());
```

Comment:
```suggestion
                                    upgraded.extensions_mut().extend(req.extensions());
```

shouldn't this just work without having to clone, given you will iterate internally and extend? Feels a bit pointless to have to clone the vec first as well? Or what am I missing?